api/src/main/java/org/hawkular/alerts/api/model/trigger/Trigger.java

```diff
@@ -283,8 +283,8 @@ public Trigger(Trigger trigger) {
         this.source = trigger.getSource();
         this.severity = trigger.getSeverity();
 
-        this.match = trigger.getMatch();
-        this.mode = trigger.getMode();
+        this.match = trigger.getMatch() != null ? trigger.getMatch() : Match.ALL;
+        this.mode = trigger.getMode() != null ? trigger.getMode() : Mode.FIRING;
```

Comment:
Probably it won't hit that logic, as when this constructor is invoked it should initialize the this.mode.
(As it happens in the other constructors).
But I think the suggestion is good, so I will update it and rebase the other PR.